File: BabylonProject/Assets/Content/Mobile/MobileShadowMaterial.cs
using System;
using UnityEditor;
using UnityEngine;

/**
* Editor Script Component
* @class MobileShadowMaterial
*/
[Babylon(Class="PROJECT.MobileShadowMaterial"), AddComponentMenu(CanvasToolsStatics.CANVAS_TOOLS_MENU + "/Content Components/Mobile/Mobile Shadow Material", 2104)]
public class MobileShadowMaterial : EditorScriptComponent
{
    [Tooltip("Create new shadow only matieral")]
    [Auto] public bool createNewMaterial = true;
}

File: BabylonProject/Assets/Content/Mobile/MobileShadowMaterial.ts
module PROJECT {
    /**
    * Babylon Script Component
    * @class MobileShadowMaterial
    */
    export class MobileShadowMaterial extends BABYLON.Toolkit.ScriptComponent {
        private createNewMaterial: boolean = true;

        protected awake(): void {
            if (this.createNewMaterial === true && this.transform instanceof BABYLON.AbstractMesh) {
                const materialName: string = this.transform.name.replace(" ", "") + ".ShadowMaterial";
                this.transform.material = new BABYLON.ShadowOnlyMaterial(materialName, this.scene);
            }
        }

        protected destroy(): void {
            if (this.createNewMaterial === true && this.transform instanceof BABYLON.AbstractMesh) {
                if (this.transform.material != null && this.transform.material instanceof BABYLON.ShadowOnlyMaterial) {
                    this.transform.material.dispose();
                }
            }
        }
    }
}